docs/hooks/use-submit.md

```diff
@@ -78,6 +78,48 @@ formData.append("cheese", "gouda");
 submit(formData);
 ```
 
+### Payload Serialization
+
+You may also submit raw JSON to your `action` and the default behavior will be to encode the key/values into `FormData`:
+
+```tsx
+let obj = { key: "value" };
+submit(obj); // -> request.formData()
+```
+
+You may also choose which type of serialization you'd like via the `encType` option:
+
+```tsx
+let obj = { key: "value" };
+submit(obj, encType: 'application/x-www-form-urlencoded'); // -> request.formData()
+```
+
+```tsx
+let obj = { key: "value" };
+submit(obj, { encType: "application/json" }); // -> request.json()
+```
+
+```tsx
+let text = "Plain ol' text";
+submit(obj, { encType: "text/plain" }); // -> request.text()
+```
+
+<docs-warn>In future versions of React Router, the default behavior will not serialize raw JSON payloads. If you are submitting raw JSON today it's recommended to specify an explicit `encType`.</docs-warn>
```

Comment:
Have we talked about making this a future flag?

Reply:
Ryan and I talked through this a good bit and landed on a heuristic that I need to get added to our docs and specifically [this diagram](https://reactrouter.com/en/main/guides/api-development-strategy#example-new-feature-flow).

A future flag is needed when there is no way to opt-in to the new (breaking) behavior at the call-site (locally), so we need to give them a config (global) opt-in flag.  For example:

* Deprecating `useTransition` in favor of `useNavigation` - this was a breaking change but they can opt-into the new behavior by using the new hook.  So we didn't need a future flag and instead can just log a deprecation warning when they call `useTransition`.
* Removing `fetcher.type`/`fetcher.submission` - this was a breaking change but they can opt-into the new behavior by changing their code to look at `fetcher.state` instead.  So again no future flag needed and instead can just log a deprecation warning when they access `fetcher.type`/`fetcher.submission`.
* v1 -> v2 route conventions - There's no way they can opt into this in the `routes/` directory so we needed to provide a future flag.
* Normalizing `fetcher.formMethod` from `post -> POST` - there was no way they could do this at the call site using existing APIs so we had to give them a global future flag.

So tyhat diagram needs a new path for breaking changes where we say "local opt-in possible?" and if so we can just log deprecation warnings and don't need to introduce a future flag.